mlperf_logging/package_checker/package_checker.py

```diff
@@ -82,21 +94,27 @@ def check_training_result_files(folder, usage, ruleset, quiet, werror,
             print('System {}'.format(system))
             print('Benchmark {}'.format(benchmark))
 
-            # The number of result files must be an exact number.
-            # Print a comprehensive message if some files in results
-            # directory do not match naming convention (results_*.txt)
-            if len(result_files) != benchmark_file_counts[benchmark]:
-                print('Expected at least {} runs, but detected {} runs.'.format(
-                    benchmark_file_counts[benchmark],
-                    len(result_files),
-                ))
-                too_many_errors = True
-                if len(all_files) > 0:
-                    print(all_files)
-                    print('Detected {} total files in directory {}, but some do not conform '
-                        'to naming convention, should you rename them to result_*.txt ?'.format(
-                        len(all_files), benchmark_folder,
+            if is_weak_scaling:
```

Comment:
Why are the rules different for weak scaling here? What are you going to do if someone submits 2N files instead of N? Which ones are you going to read?

Reply:
In weak scaling, there is no upper bound on the number of result files. If someone submit 2N files, that is fine and all of those files are read.

The idea of weak scaling is to evaluate the "total training capacity" of a system. Given a system with T accelerators, how long does it take to train all of the M models, where each model needs S accelerators. M can be any number bigger than N. In fact, if we fix the time budget, the higher the M and S are, the bigger training capacity the system has.

https://github.com/mlcommons/training_policies/blob/master/hpc_training_rules.adoc#62-weak-scaling-throughput